server.go

```diff
@@ -1331,6 +1319,49 @@ func (s *Server) WriteSeries(database, retentionPolicy string, points []Point) (
 		retentionPolicy = rp.Name
 	}
 
+	// Collect responses for each channel.
+	type resp struct {
+		index uint64
+		err   error
+	}
+	ch := make(chan resp, len(points))
+
+	// Write each point in parallel.
+	var wg sync.WaitGroup
+	for i := range points {
+		wg.Add(1)
+		go func(p *Point) {
+			index, err := s.writePoint(database, retentionPolicy, p)
+			ch <- resp{index, err}
+			wg.Done()
+		}(&points[i])
+	}
+	wg.Wait()
+	close(ch)
```

Comment:
That's a good point. This is just a stop gap solution while we work on more robust `Server`-level batching.